### ISSUE TITLE
Include light font-weights in contrast check

### ISSUE THREAD

**Issue by jyasskin** [dequelabs/axe-core#627]
The WCAG guidelines say that bold font-weights allow lower contrast to be accessible, but they don't currently say that light font-weights need higher contrast to be accessible, even though https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html does mention "very thin fonts". Would it make sense to warn for contrasts less than X (7?) if the font-weight is 300 or lower, in the same way you allow contrasts down to 3 if the font-weight is 600 or above?

**Comment by alice** [comment]
And/or, could we consider that low font weight is itself akin to low contrast, and only allow it on larger font sizes?

**Comment by WilcoFiers** [comment]
It's an interesting thought. I don't think we should do this as a WCAG rule, but as a best practice it makes a lot of sense. Are either of you interested in working on a rule like this?

**Comment by jyasskin** [comment]
I agree that it's not a WCAG rule. I may have some time for this tomorrow. Are you happy with Alice's suggestion? That'd be, "if font-weight<=300 and (contrast <= 4.5 or the font's small), fail"?

**Comment by alice** [comment]
+1 - effectively, low font weight requires *both* large font size *and* AAA contrast (with size taken into account).

**Comment by WilcoFiers** [comment]
Sounds good to me! Thanks for working on this. Let me know if you have any questions.

**Comment by marcysutton** [comment]
Hi all, just to follow up–are you still interested in working on this change? We would love to get a PR for it.

**Comment by WilcoFiers** [comment]
Closing this issue due to inactivity. If someone is still interested in writing this rule, feel free to open a new issue or a pull request.
